Common/src/main/java/com/blamejared/crafttweaker/api/bracket/custom/TagManagerBracketHandler.java

```
@@ -18,63 +22,113 @@
+        if(elementClass.isPresent()) {
+            String elementName = CraftTweakerAPI.getRegistry()
+                    .getZenClassRegistry()
+                    .getNameFor(loader, elementClass.get())
+                    .orElseThrow(() -> new ParseException(position, "Unable to make tag for unknown class!"));
+            managerZCName += "<" + elementName + ">";
+        }
         
         
-        final String implementationZCTypeFor = tagRegistry.getImplementationZCTypeFor(tagFolder);
-        final List<IParsedType> typeArguments = Collections.singletonList(ParseUtil.readParsedType(implementationZCTypeFor, position));
-        final ParsedCallArguments arguments = new ParsedCallArguments(typeArguments, Collections.emptyList());
+        final List<IParsedType> typeArguments = Collections.singletonList(ParseUtil.readParsedType(managerZCName, position));
+        final ParsedCallArguments arguments = new ParsedCallArguments(typeArguments, List.of(ParseUtil.createResourceLocationArgument(position, manager.resourceKey()
+                .location())));
         
         return new ParsedExpressionCall(position, member, arguments);
     }
     
-    private ParsedExpression createCallSynthetic(String tagFolder, CodePosition position) {
+    static void confirmTagFolderExists(String tagFolder, CodePosition position) throws ParseException {
         
-        final ParsedExpressionVariable tags = new ParsedExpressionVariable(position, CrTTagRegistry.GLOBAL_NAME, null);
-        final ParsedExpressionMember member = new ParsedExpressionMember(position, tags, "getForElementType", null);
-        final String type = tagRegistry.getElementZCTypeFor(tagFolder);
-        final IParsedType elementType = ParseUtil.readParsedType(type, position);
-        final List<IParsedType> typeParam = Collections.singletonList(elementType);
+        try {
+            ResourceLocation location = new ResourceLocation(tagFolder);
```

Just food for thought 😉 

Try-catch vs. `tryParse()` method or smth?

---

In this case we want to catch and rethrow the exception instead of silently swallowing it.